### ISSUE TITLE
Print in ODE when stop condition is reached

### ISSUE THREAD

**Issue by tchamelot** [Axect/Peroxide#52]
Hello, thank your for this nice crate.
I have a minor issue that is not blocking or whatever. In the ODE method `integrate` there  are `printl` when the stop condition is reached. I can understand why for debug purpose but when calling this method a lot of time, this end up making a lot of noise and slowing down the process.
For example, my use case is a physical model being simulated several times to find the best parameters for a PID controller through black box optimization (differential evolution). The stop condition is used to eliminate parameters that do not satisfy the physical limits of my model (a motor that spins faster than the maximum angular speed allowed by the voltage supply for example). I end up with thousands of line printed by the stop condition which slow the process and make the interesting output difficult to identify.
I would be up to write a small patch to fix this but i do not know how yet.

The print is at this line:
https://github.com/Axect/Peroxide/blob/892e224b4bfa997238203d1d677184a647d5e94b/src/numerical/ode.rs#L607

**Comment by Axect** [comment]
@tchamelot Thanks for the good point. The initial purpose of stop condition was to stop when the error falls below a certain level or when the score measured by a specific metric exceeds a certain level. Here is [the example](https://github.com/Axect/Peroxide_Gallery/tree/master/Numeric/rk4_with_plot).

```rust
fn main() {
    let init_state = State::<f64>::new(0f64, c!(1), c!(0));

    let mut ode_solver = ExplicitODE::new(test_fn);

    ode_solver
        .set_method(ExMethod::RK4)
        .set_initial_condition(init_state)
        .set_step_size(0.01)
        .set_stop_condition(stop)        // Add stop condition
        .set_times(1000);

	// omit unnecessary parts
}

fn test_fn(st: &mut State<f64>) {
    // some functions
}

fn stop(st: &ExplicitODE) -> bool {
    let y = &st.get_state().value[0];
    (*y - 2.4).abs() < 0.01
}
```

However, I agree that using `println` inside solve function is too inefficient. Therefore, it would be better to include a variable and create a method (e.g. `fn has_stopped(&self) -> bool`) that can determine whether a stop condition is reached after `integrate`.

I'd appreciate it if you could let me know what you think about this (Is this method ok?, what method name would be good if it's okay, etc.)

**Comment by tchamelot** [comment]
Thank you for the answer, I have begun to look into it.
When the stop condition hits i currently identify it through the record size. In the example you gave, i would except the record to contain 1001 rows without the stop condition. When the stop condition hits, i except a record with less than 1001 rows.
I don't know if this is enough. I can also add a `fn has_stopped(&self) -> bool` function as you suggest (the function name is great). The information would be redundant but it would be easier for the end user.
I will go with the additional function and create a PR soon.

**Comment by Axect** [comment]
I published Ver `0.31.8` including your PR :rocket:
Now, you can escape from so many prints!

**Comment by tchamelot** [comment]
Thank you :)
